### DISCUSSION TITLE
Any way to show what files golangci-lint has been run against?

### DISCUSSION THREAD

**Discussion by veqryn** [golangci/golangci-lint#4228] [Q&A]
I have a weird situation where `golangci-lint run ./...` is showing no problems, but `golangci-lint run ./submodule/cmd/main.go` is showing lots of errors.

I have tried using the `--verbose` flag, but it doesn't show much of anything.
I couldn't figure out what format `--trace-path` outputs in, and i'm not sure what info it contains.

Is there any way to have a run against `./...` output all the files being considered or looked at?

**Comment by ldez** [comment]
Hello,

Which version are you using?
Can you provide your configuration?
Can you provide the real file tree?

Edit: a comment has been removed; this message was related to it.

**Comment by veqryn** [comment]
Version: golangci/golangci-lint-action@v3
https://github.com/veqryn/slog-context/blob/main/.github/workflows/golangci-lint.yml#L32

Configuration: https://github.com/veqryn/slog-context/blob/main/.golangci.yml

Repo and file structure: https://github.com/veqryn/slog-context

**Reply by ldez** [reply]
Thank you. Why are you running golangci-lint twice?

Note: You are using a very old version of the GitHub action

**Reply by ldez** [reply]
Do you still have the same problem?

**Reply by ldez** [reply]
I think I know what your problem was: the files inside the directory `examples` were ignored, am I right?

**Reply by ldez** [reply]
My guess was based on old v1 default exclusion rules, but `examples` is a sub-module, so this is not related.

So I need more information about the files that are ignored

**Reply by ldez** [reply]
Ok, I think your problem was submodules.

golangci-lint should be executed inside the module (or the submodule) to be able to analyze it.

**Reply by veqryn** [reply]
Yes, that is the problem.  golangci-lint does not appear to see submodules.
This is a problem because it is not obvious at all.
If I run a command that ends in `./...` I expect it to be run against everything in this folder and recursively in any sub-directories.
golangci-lint should either run itself against submodules discovered in sub-directories, or it should produce a warning at the very least that a submodule was detected but NOT run against.

**Reply by ldez** [reply]
golangci-lint cannot see submodules, so it cannot produce a warning.
golangci-lint can only be run on a module.

This is the same thing as `go build`

**Reply by veqryn** [reply]
Could that be a feature request then? To check for submodules?

**Reply by ldez** [reply]
The Go tooling for static analysis doesn't allow that.

**Reply by veqryn** [reply]
golangci-lint is an executable right? It could read the file directories looking for go.mod files, before it hooks into the static analysis built in tool?

**Reply by ldez** [reply]
`go build`, `go test`, `go vet`, etc. don't do that, so we will not do it too, sorry.
